runtime/src/vote_account.rs

```diff
@@ -92,16 +204,104 @@ impl PartialEq<VoteAccount> for VoteAccount {
     }
 }
 
+impl Default for VoteAccounts {
+    fn default() -> Self {
+        Self {
+            vote_accounts: HashMap::default(),
+            staked_nodes: RwLock::default(),
+            staked_nodes_once: Once::new(),
+        }
+    }
+}
+
+impl Clone for VoteAccounts {
+    fn clone(&self) -> Self {
+        if self.staked_nodes_once.is_completed() {
+            let staked_nodes = self.staked_nodes.read().unwrap().clone();
+            let other = Self {
+                vote_accounts: self.vote_accounts.clone(),
+                staked_nodes: RwLock::new(staked_nodes),
+                staked_nodes_once: Once::new(),
+            };
+            other.staked_nodes_once.call_once(|| {});
+            other
+        } else {
+            Self {
+                vote_accounts: self.vote_accounts.clone(),
+                staked_nodes: RwLock::default(),
+                staked_nodes_once: Once::new(),
+            }
+        }
+    }
+}
+
+impl PartialEq<VoteAccounts> for VoteAccounts {
+    fn eq(&self, other: &Self) -> bool {
+        self.vote_accounts == other.vote_accounts
+    }
+}
+
+type VoteAccountsHashMap = HashMap<Pubkey, (u64 /*stake*/, ArcVoteAccount)>;
+
+impl From<VoteAccountsHashMap> for VoteAccounts {
+    fn from(vote_accounts: VoteAccountsHashMap) -> Self {
+        Self {
+            vote_accounts,
+            staked_nodes: RwLock::default(),
+            staked_nodes_once: Once::new(),
+        }
+    }
+}
+
+impl Borrow<VoteAccountsHashMap> for VoteAccounts {
+    fn borrow(&self) -> &VoteAccountsHashMap {
+        &self.vote_accounts
+    }
+}
+
+impl FromIterator<(Pubkey, (u64 /*stake*/, ArcVoteAccount))> for VoteAccounts {
+    fn from_iter<I>(iter: I) -> Self
+    where
+        I: IntoIterator<Item = (Pubkey, (u64, ArcVoteAccount))>,
+    {
+        Self::from(HashMap::from_iter(iter))
+    }
+}
+
+impl Serialize for VoteAccounts {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.vote_accounts.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for VoteAccounts {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let vote_accounts = VoteAccountsHashMap::deserialize(deserializer)?;
+        Ok(Self::from(vote_accounts))
+    }
+}
```

Comment:
@sakridge regarding the backward compatibility. This will serialize/deserialize as a
```
HashMap<Pubkey, (u64, ArcVoteAccount)>
```
which was is the current type in the `Stakes` struct:
https://github.com/solana-labs/solana/blob/79fb64687/runtime/src/stakes.rs#L13